packages/swingset-liveslots/src/boyd-gc.js

```diff
@@ -0,0 +1,438 @@
+import { parseVatSlot } from './parseVatSlots.js';
+
+/*
+  Theory of Operation (imports/Presences)
+
+    Imports are in one of 5 states: UNKNOWN, REACHABLE, UNREACHABLE,
+    COLLECTED, FINALIZED. Note that there's no actual state machine with those
+    values, and we can't observe all of the transitions from JavaScript, but
+    we can describe what operations could cause a transition, and what our
+    observations allow us to deduce about the state:
+
+    * UNKNOWN moves to REACHABLE when a crank introduces a new import
+    * userspace holds a reference only in REACHABLE
+    * REACHABLE moves to UNREACHABLE only during a userspace crank
+    * UNREACHABLE moves to COLLECTED when GC runs, which queues the finalizer
```

Comment:
Probing my knowledge:
```suggestion
    * UNREACHABLE moves to COLLECTED when engine GC runs, which queues the finalizer
```